controllers/metal3datatemplate_controller.go

```diff
@@ -166,10 +164,47 @@ func (r *Metal3DataTemplateReconciler) reconcileDelete(ctx context.Context,
 func (r *Metal3DataTemplateReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&capm3.Metal3DataTemplate{}).
+		Watches(
+			&source.Kind{Type: &capm3.Metal3DataClaim{}},
+			&handler.EnqueueRequestsFromMapFunc{
+				ToRequests: handler.ToRequestsFunc(r.Metal3DataClaimToMetal3DataTemplate),
+			},
+			// Do not trigger a reconciliation on updates of the claim, as the Spec
+			// fields are immutable
+			builder.WithPredicates(predicate.Funcs{
+				UpdateFunc:  func(e event.UpdateEvent) bool { return false },
```

Comment:
That is because we do not want to trigger a reconciliation if the claim changes. With the webhooks, the fields that matter, like the template, are immutable, meaning that an update on the claim object won't change anything that would be used in the reconciliation loop. so we don't need to trigger on updates.